website_sale_stock_list_preview/static/src/js/website_sale_stock_list_preview.js

```diff
@@ -47,6 +47,16 @@ odoo.define("website_sale_stock_list_preview.shop_stock", function(require) {
                                 )
                             ).get(0)
                         );
+                    if (
+                        product.virtual_available == 0 &&
+                        (product.inventory_availability == "always" ||
+                            product.inventory_availability == "threshold")
+                    ) {
+                        $(product_dic[product.id])
+                            .find(".fa-shopping-cart")
```

Comment:
Is there a client qweb template that displays this? Can you add a `t-if` in that template?

Reply:
@Yajo What about of load performance....¿?¿?¿??¿

Reply:
It depends on the specific case. May you point me to where the shopping cart icon is rendered? This way I can see if there's a better way to do this.

In any case, consider this: if the availability is being lazily-loaded as I see here (which is OK if that takes long), there's a possible race condition where a user could click the cart icon before it gets hidden. Wouldn't it be better to hide it before, and just show it later here if there's stock? :thinking: